### ISSUE TITLE
shellcheck warns against required word splitting.

### ISSUE THREAD

**Issue by orbea** [koalaman/shellcheck#1002]
#### For bugs
- Rule Id: `SC2046` and `SC2086`
- My shellcheck version: `online`
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

set -eu

foo='ls -l'
bar='/tmp/dir'

mkdir -p -- "$bar"

command -p $foo "$bar"
command -p $(printf %s "$foo") "$bar"

```

#### Here's what shellcheck currently says:

```
Line 10:
command -p $foo "$bar"
           ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 11:
command -p $(printf %s "$foo") "$bar"
           ^-- SC2046: Quote this to prevent word splitting.
```

#### Here's what I wanted or expected to see:

Nothing.

This show two variations of the same concept, when dynamically running commands with `command` the actual command should not be quoted or it will result in an error like `ls -l: command not found`. In cases like this the destination path should be not part of the command variable should be in quotes to account for cases when it contains spaces.  Additionally the second variation with `printf` seems to be required for ksh on my system.

**Comment by kurahaupo** [comment]
There are other ways this do not rely on word splitting; consider using them instead. (Arrays, functions.)

It is perfectly valid to disable Shellcheck's warnings; they exist to catch naïve errors, not to demand conformance.

**Comment by orbea** [comment]
With limited shells I do not have access to bash arrays and a function may work for this simplified example, but there are far more elaborate examples in scripts where a function would not help quite as easily.

Consider this.
```
#!/bin/sh

set -eu

cmd () {
  command=; print=
  while [ "$#" -gt 1 ]; do
    var="$1"; shift
    command="${command} $var"
    case "$var" in
      -- ) continue ;;
      * ) print="${print} $var" ;;
    esac
  done
  printf %s\\n " $print '$1'"
  command -p $(printf %s "${command#* }") "$1"
}

bar='/tmp/dir'

cmd mkdir -p -- "$bar"
cmd ls -l -- "$bar"
```
Output.
```
  mkdir -p '/tmp/dir'
  ls -l '/tmp/dir'
```
I understood shellcheck intends to lint shell scripts and incorrect warnings can only make people lose confidence in its ability to do just that.

**Comment by orbea** [comment]
I have come up with a work around to silence this warning. Here is an updated example.
```
#!/bin/sh

set -eu

foo='ls -l'
bar='/tmp/dir'

mkdir -p -- "$bar"

eval "set -- $foo"
command -p "$@" "$bar"
```

**Comment by orbea** [comment]
Note, I had difficulty applying the above to my more elaborate example until I added this to the top the script.
```
PATH="$(command -p getconf PATH):$PATH"
```
For some reason ksh did not like this with `command -p`, but was fine with just `command`.
